### ISSUE TITLE
Basic AuthException thrown instead of UserNotConfirmedException on Signin

### ISSUE THREAD

**Issue by cto-leaps** [aws-amplify/amplify-flutter#577]
**Describe the bug**
I'm trying to catch an unconfirmed user upon Signing in.
I do get an AuthException (message: User is not confirmed), but not the expected UserNotConfirmedException.
So it's making it hard for me to redirect to the Code Verification screen when needed.

**To Reproduce**
1. Create an account but don't proceed to the Code entry
2. Restart App, try to sign in
3. A basic AuthException in thrown, not the expected UserNotConfirmedException

**Expected behavior**
UserNotConfirmedException should be thrown

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[] Android
[] iOS

<details>

  <summary>Output of <code>flutter doctor -v</code></summary>
  
  ```[✓] Flutter (Channel stable, 2.0.6, on macOS 11.2.3 20D91
    darwin-arm, locale en-FR)
    • Flutter version 2.0.6 at
      /Users/erwanjessonripoll/Dev/flutter
    • Framework revision 1d9032c7e1 (2 weeks ago),
      2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3

[!] Android toolchain - develop for Android devices
    (Android SDK version 30.0.3)
    • Android SDK at
      /Users/erwanjessonripoll/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    ! Some Android licenses not accepted.  To resolve
      this, run: flutter doctor --android-licenses

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google
      Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.56.0)
    • VS Code at /Applications/Visual Studio
      Code.app/Contents
    • Flutter extension version 3.22.0

[✓] Connected device (3 available)
    • iPhone 12 (mobile) • BB19A73D-4601-4312-9C96-4DB68D0F7C8D •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-5
      (simulator)
    • macOS (desktop)    • macos                                •
      darwin-arm64   • macOS 11.2.3 20D91 darwin-arm
    • Chrome (web)       • chrome                               •
      web-javascript • Google Chrome 90.0.4430.212```

</details>


**Smartphone (please complete the following information):**
 - Device: Simulator and Physical device iPhone 12 Pro Max
 - OS: iOS 14.5

**Additional context**
  amplify_flutter: 0.1.4
  amplify_auth_cognito: 0.1.4
  amplify_analytics_pinpoint: 0.1.4
  amplify_api: 0.1.4
  amplify_storage_s3: 0.1.4

AWSAuthCore 2.24.1 
AWSCognitoIdentityProvider 2.24.1
AWSCognitoIdentityProviderASF 2.24.1
AWSCore 2.24.1
AWSMobileClient 2.24.1
AWSPinpoint 2.24.1
AWSPluginsCore 1.9.2
AWSS3 2.24.1
Amplify 1.9.2 
AmplifyPlugins 1.9.2
AppSyncRealTimeClient 1.5.0

**Comment by ragingsquirrel3** [comment]
I had a quick look at this and was able to reproduce on iPhone. Android seems to throw `UserNotConfirmedException`. I will continue investigating.

**Comment by cto-leaps** [comment]
Thanks @ragingsquirrel3 !
Much appreciated!

**Comment by ragingsquirrel3** [comment]
@cto-leaps thanks for bringing this to our attention. A fix has been merged to master and will go out with the next release.

**Comment by cto-leaps** [comment]
@ragingsquirrel3 Many thanks!!!
Any idea when this will be pushed?

**Comment by rmargolinross** [comment]
Will this be in the null safety release or be part of 0.1.6, if that is coming out?

**Comment by Jordan-Nelson** [comment]
@cto-leaps and @rmargolinross - This should be fixed as of 0.1.6. Please let us know if you are still experiencing this on 0.1.6.

**Comment by offlineprogrammer** [comment]
Hi All

I am closing this issue for now We can reopen it if you are still facing the issue and when you provide details

Regards
Mo
